guide/src/ref/packet-log.md

```diff
@@ -1,37 +1,167 @@
+Notice that, the Python script can be slower than the Rust subcommand for one to
+two orders of magnitude when processing large log files.
+
+Example usage:
+
+```bash
+python scripts/log_converter.py ./packet.rtl ./output
+```
+
+The paths to input and output files are the same as those for `rattan convert`.
+
+The behaviour of the python script is basically the same as `rattan convert`,
+except for that the output path prefix is not optional.
+
 ## Specification
 
-The packet log is stored in system endianness (little-endian in most cases and in our tests), entry by entry. The specification for each log entry is as follows:
+### Raw file
+
+A `.raw` file is an unstructed binary file. Currently, the headers recorded
+in `raw-tcp` and `raw-ip` modes are directly written into the `.raw` file continuously,
+without any padding, header or index.
+The file is considered as a byte buffer, that a slice (starts from `offset` Bytes since
+the head of the file and ends at `offset` + `length` ),
+in it can be indexed by the pair (`offset`, `length`).
+
+### Log entry file
+
+Now we have 2 kinds of log entry files:
+
+- `.flows` files, which contains the metadata for flows. They consist of `Flow Entries`.
+- `.rtl` files, which contains per-packet log for packets in those flows.
+  - For non-raw modes, they consist of non-raw `General Packet Entries`.
+  - For now modes, they consist of chunks. Each chunk starts with a `Chunk Prologue`, which is followed by raw `General Packet Entries`.
+
+Above mentioned log entry types are described as follows.
+
+Every log entry file consists of `Log Entries`.
+Those log entries are stored in system endianness (little-endian in most cases and in our tests).
+The specification for a log entry is as follows:
+
+```text
+0                   1                   2                   3
+0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|       LH.length       | LH.ty.|                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               +
+|                                                               |
++                                                               +
+|                            payload                            |
++                                                               +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
+```bash
+protocol "LH.length:12,LH.ty.:4,payload:112"
+```
+
+Generated with [protocol](https://github.com/luismartingarcia/protocol), where:
+
+- `LH` is short for log entry header
+
+`LH.length` denotes the whole length of the log entry file, including the header(2 Bytes) and its payload, in Bytes.
+
+Now there are mutiple possibility of `LH.ty.` that decides how the payload of an
+log entry should be interpreted. They are:
+
+- `LH.ty` = 0, for a General Packet Entry as payload.
```

Comment:
You should always keep the expression in the same format. You use `LH.ty.` above, you shouldn't use `LH.ty` here, instead, you should still use `LH.ty.`